documentation/puppet/v3/environment_classes.markdown

```diff
@@ -0,0 +1,294 @@
+  example, if an "Integer" type class parameter were defined in the manifest as
+  having a default value of "3", the "default_literal" element for the
+  parameter will contain a JSON Number type of 3.
+
+* The environment classes API does not provide a way to filter the list of
+  classes returned via use of a search string.  The environment classes API
+  returns information for all classes found within an environment's manifest
+  files.
+
+* Unlike the resource type API under Puppet 4, the environment classes API
+  does include the filename in which each class was found.  The resource 
+  type API under Puppet 3 does include the filename but the resource type 
+  API under Puppet 4 does not.
+
+* The environment classes API does not include the line number at which a
+  class is found in the file.
+
+* Unlike the resource types API under Puppet 3, the environment classes API 
+  does not include any doc strings for a class entry.  Note that doc strings 
+  are also not returned for class entries in the Puppet 4 resource type API.
+  
+* The environment classes API returns a file entry for manifests that exist 
+  in the environment but in which no classes were found.  The resource type
+  API omits entries for files which did not contain any classes.
+
+* The Content-Type in the response to an environment classes API query is 
+  "application/json" whereas the resource types API uses a Content-Type of 
+  "text/pson".
+  
+* If an error is encountered when parsing a manifest, the resource types API 
+  omits information from the manifest entirely - although it will, 
+  assuming none of the parsing errors were found in one of the files associated
+  with the environment's
+  [`manifest` setting](/puppet/latest/reference/config_file_environment.html#manifest),
+  include class information from other manifests that could successfully be
+  parsed.  In the case that one or more classes is returned but errors were 
+  encountered parsing other manifests, the response from the resource types API 
+  call does not include any explicit indication that a parsing error has been
+  encountered.  The environment classes API will include information for every
+  class that can successfully be parsed.  For any errors which occur when 
+  parsing individual manifest files, an entry for the corresponding manifest 
+  file is included, along with an "error" with a detail string about the failure
+  encountered during parsing.
+
+## `GET /puppet/v3/environment_classes?environment=:environment`
+
+Making a request with no query parameters is currently not supported and will
+provide an HTTP 400 (Bad Request) response.
+
+### Supported HTTP Methods
+
+GET
+
+### Supported Formats
+
+JSON
+
+### Query Parameters
+
+One parameter should be provided to the GET:
+
+* `environment`: Only the classes and parameter information pertaining to the
+  specified environment will be returned for the call.
+
+### Responses
+
+#### Get With Results
+
+~~~~
```

Comment:
This doesn't need to block merging this PR, but you might be happy to know the docs site now supports triple backticks like the rest of the modern world. :dancer: